quic/s2n-quic-core/src/transport/parameters/mod.rs

```diff
@@ -853,13 +853,18 @@ impl TransportParameterValidator for MaxAckDelay {
 //#    active connection migration (Section 9) on the address being used
 //#    during the handshake.
 
-#[derive(Clone, Copy, Debug, Default, PartialEq)]
+#[derive(Clone, Copy, Debug, PartialEq)]
 pub enum MigrationSupport {
-    #[default]
     Enabled,
     Disabled,
 }
 
+impl MigrationSupport {
+    pub const fn default() -> Self {
+        MigrationSupport::Enabled
+    }
+}
+
 impl TransportParameter for MigrationSupport {
     type CodecValue = ();
```

Comment:
Is this just a refactor or was there a need to do this?

Reply:
I needed it to be `const` for use in this function: https://github.com/aws/s2n-quic/pull/2182/files#diff-7b125c20a9c47acef408f6eb14de8c464499618fa4aaf74f0ea3413d7efedd72R93

Reply:
I would prefer to preserve the `Default` implementation, since that's really common to use. Then we can do something similar to the others with associated `const RECOMMENDED: Self`